### ISSUE TITLE
False warning CA2248

### ISSUE THREAD

**Issue by NN---** [dotnet/roslyn-analyzers#4711]
### Analyzer

**Diagnostic ID**: [CA2248](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2248): `Do not use ReferenceEquals with value types`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.102](https://dotnet.microsoft.com/download/dotnet/5.0)

CA2248 says the call returns `false`, but this program prints `True`.

```cs
using System;

enum A { Z = 0 }

internal class Program
{
    static void Main()
    {
        Console.Write(A.Z.HasFlag(A.Z)); // warning CA2248: This call will always returns 'false' because the enum type 'A' is not marked with 'FlagsAttribute'
    }
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <Nullable>enable</Nullable>
    <AnalysisLevel>latest</AnalysisLevel>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
  </PropertyGroup>

</Project>
```

**Comment by Youssef1313** [comment]
Probably fixed in #4575

**Comment by mavasani** [comment]
Yes, confirmed this is fixed with https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.3. Please also remove `<EnableNETAnalyzers>true</EnableNETAnalyzers>` when using the NuGet package, as that property renders the NuGet package redundant causing built in analyzers in the SDK to be preferred over the ones from the NuGet package.
